File: src/main/java/com/braggbnbsevenseventyseven/service/GuestService.java
package com.braggbnbsevenseventyseven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnbsevenseventyseven.domain.Guest;
import com.braggbnbsevenseventyseven.dto.GuestDTO;
import com.braggbnbsevenseventyseven.dto.GuestSearchDTO;
import com.braggbnbsevenseventyseven.dto.GuestPageDTO;
import com.braggbnbsevenseventyseven.dto.GuestConvertCriteriaDTO;
import com.braggbnbsevenseventyseven.service.GenericService;
import com.braggbnbsevenseventyseven.dto.common.RequestDTO;
import com.braggbnbsevenseventyseven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface GuestService extends GenericService<Guest, Integer> {

	List<Guest> findAll();

	ResultDTO addGuest(GuestDTO guestDTO, RequestDTO requestDTO);

	ResultDTO updateGuest(GuestDTO guestDTO, RequestDTO requestDTO);

    Page<Guest> getAllGuests(Pageable pageable);

    Page<Guest> getAllGuests(Specification<Guest> spec, Pageable pageable);

	ResponseEntity<GuestPageDTO> getGuests(GuestSearchDTO guestSearchDTO);
	
	List<GuestDTO> convertGuestsToGuestDTOs(List<Guest> guests, GuestConvertCriteriaDTO convertCriteria);

	GuestDTO getGuestDTOById(Integer guestId);







}

File: braggbnbsevenseventyseven_ui/src/services/GuestService.js
import http from "../http-common"; 

class GuestService {
  getAllGuests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/guest/guests`, searchDTO);
  }

  get(guestId) {
    return this.getRequest(`/guest/${guestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/guest?field=${matchData}`, null);
  }

  addGuest(data) {
    return http.post("/guest/addGuest", data);
  }

  update(data) {
  	return http.post("/guest/updateGuest", data);
  }
  
  uploadImage(data,guestId) {
  	return http.postForm("/guest/uploadImage/"+guestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new GuestService();
